addons/storyshots/storyshots-core/src/frameworks/configure.ts

```diff
@@ -1,27 +1,143 @@
+  files: string[];
+}
+
+function matches(storyKey: string, arrayOrRegex: string[] | RegExp) {
+  if (Array.isArray(arrayOrRegex)) {
+    return arrayOrRegex.includes(storyKey);
+  }
+  return storyKey.match(arrayOrRegex);
+}
+
+function isExportStory(key: string, { includeStories, excludeStories }: any) {
+  return (
+    // https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs
+    key !== '__esModule' &&
+    (!includeStories || matches(key, includeStories)) &&
+    (!excludeStories || !matches(key, excludeStories))
+  );
+}
+
+const loadStories = (loadable: string, framework: string, clientApi: ClientApi) => {
+  const currentExports = new Map();
+
+  // eslint-disable-next-line global-require, import/no-dynamic-require
+  const exported = require(loadable);
+
+  // An old-style story file
+  if (!exported.default) {
+    return;
+  }
 
-  if (fs.lstatSync(resolvedConfigPath).isDirectory()) {
-    return path.join(resolvedConfigPath, 'config');
+  if (!exported.default.title) {
+    throw new Error(`Unexpected default export without title: ${JSON.stringify(exported.default)}`);
   }
 
-  return resolvedConfigPath;
+  const { default: meta, ...exports } = exported;
+  const { title: kindName, parameters: params, decorators: decos, component } = meta;
+
+  // @ts-ignore We pass true here to avoid the warning about HMR. It's cool clientApi, we got this
+  const kind = clientApi.storiesOf(kindName, true);
+
+  // we should always have a framework, rest optional
+  kind.addParameters({
+    framework,
+    component,
+    fileName: currentExports.get(exported),
+    ...params,
+  });
+
+  (decos || []).forEach((decorator: DecoratorFunction) => {
+    kind.addDecorator(decorator);
+  });
+
+  Object.keys(exports).forEach(key => {
+    if (isExportStory(key, meta)) {
+      const storyFn = exports[key];
+      const { name, parameters, decorators } = storyFn.story || {};
+      const decoratorParams = decorators ? { decorators } : null;
+      const displayNameParams = name ? { displayName: name } : {};
+      kind.add(storyNameFromExport(key), storyFn, {
+        ...parameters,
+        ...decoratorParams,
+        ...displayNameParams,
+      });
+    }
+  });
+};
+
+function getConfigPathParts(input: string): Output {
+  const configDir = path.resolve(input);
+
+  if (fs.lstatSync(configDir).isDirectory()) {
+    const output: Output = { files: [], stories: [] };
+    const main = path.join(configDir, 'main.js');
+    const preview = path.join(configDir, 'preview.js');
+    const config = path.join(configDir, 'config.js');
+    const configTS = path.join(configDir, 'config.ts');
+
+    if (isFile(preview)) {
+      output.files.push(preview);
+    }
+    if (isFile(main)) {
+      const { stories = [] } = require.requireActual(main);
+
+      const result = stories.reduce((acc: string[], i: string) => [...acc, ...glob.sync(i)], []);
+
+      output.stories = result;
+    }
+    if (isFile(config)) {
+      output.files.push(config);
```

Comment:
Should we allow both config + preview here? (ie should this be an error?)

Reply:
done